names/names.py

```diff
@@ -11,10 +11,73 @@
 f.close()
 
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+
+names_2_values = {}
+
+for name in names_2:
+    names_2_values[name] = name
```

Comment:
you could also do this with a set. 

name_2_set = set(names_2)
the rest of your code would work here